### PR TITLE
Update DDP dynamo debug docs

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118295

Refreshes https://github.com/pytorch/pytorch/pull/114201 and updates it to include other log names that also include ddp_optimizer.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by LucasLLC — APPROVED** [review]
lgtm! thanks!

File: docs/source/notes/ddp.rst [hunk 1]
```diff
@@ -218,8 +218,8 @@ to fire in-between sections of backwards, and schedule communications to overlap
 
 See `this blog post <https://dev-discuss.pytorch.org/t/torchdynamo-update-9-making-ddp-work-with-torchdynamo/860/1>`_ for
 a more in-depth explanation and experimental results, or read the docs and code at
-`torch/_dynamo/optimizations/distributed.py <https://github.com/pytorch/pytorch/blob/4908a12542798a3e8641faae6b74f068fdfc6778/torch/_dynamo/optimizations/distributed.py#L56>`_
+`torch/_dynamo/optimizations/distributed.py <https://github.com/pytorch/pytorch/blob/bbc39b7bb48d28d67e3253a89cc82df3687ddd1b/torch/_dynamo/backends/distributed.py#L124>`_
 
-To Debug DDPOptimizer, set `torch._dynamo.config.log_level` to DEBUG (for full graph dumps) or INFO
+To Debug DDPOptimizer, set `TORCH_LOGS='ddp_graphs'` for full graph dumps. For lighter weight debug prints, add any of 'dynamo', 'distributed', or 'dist_ddp' to  `TORCH_LOGS`
```
Comment (wanchaol) [inline docs/source/notes/ddp.rst:223]: For lighter weight "use"?

**Review by wanchaol — APPROVED** [review]

**Pull request closed without merging**